custom/covid/management/commands/update_case_index_relationship.py

```diff
@@ -8,47 +8,65 @@
 from corehq.apps.hqcase.utils import submit_case_blocks
 from corehq.form_processor.interfaces.dbaccessors import CaseAccessors
 
+'''This command has an optional argument '--location' that will exclude all cases with that location. If the
+case_type is lab_result, the owner_id of that extension case is set to '-'. '''
 
 BATCH_SIZE = 100
 DEVICE_ID = __name__ + ".update_case_index_relationship"
 
 
-def should_skip(case):
-    return len(case.indices) != 1
+def should_skip(case, traveler_location_id):
+    if traveler_location_id is None:
+        return len(case.indices) != 1
+    return len(case.indices) != 1 or case.get_case_property('owner_id') == traveler_location_id
 
 
 def needs_update(case):
     index = case.indices[0]
     return index.referenced_type == "patient" and index.relationship == "child"
 
 
+def get_owner_id(case_type):
+    if case_type == 'lab_result':
+        return '-'
+    return None
+
+
 class Command(CaseUpdateCommand):
     help = ("Updates all case indices of a specfied case type to use an extension relationship instead of parent.")
 
-    def case_block(self, case):
+    def case_block(self, case, owner_id):
         index = case.indices[0]
         return ElementTree.tostring(CaseBlock.deprecated_init(
             create=False,
             case_id=case.case_id,
+            owner_id=owner_id,
             index={index.identifier: (index.referenced_type, index.referenced_id, "extension")},
         ).as_xml()).decode('utf-8')
 
     def update_cases(self, domain, case_type, user_id):
         accessor = CaseAccessors(domain)
         case_ids = accessor.get_case_ids_in_domain(case_type)
         print(f"Found {len(case_ids)} {case_type} cases in {domain}")
+        traveler_location_id = self.location
```

Comment:
Nitpick: It's weird that the superclass knows about `location` even though I think it's only used by some subclasses. I'd expect the superclass to do something more like create a `self.extra_options` dict with anything that's in `options` other than the options the superclass knows about (`and_linked`, `username`).